widgets/sitesMap/src/widget.tsx

```diff
@@ -0,0 +1,131 @@
+    id: 'sitesMap',
+    name: 'Sites Map',
+    description: 'This widget displays a map view of sites by location with site deployments status summary',
+    initialWidth: 6,
+    initialHeight: 30,
+    color: 'green',
+    isReact: true,
+    hasReadme: true,
+    hasStyle: true,
+    permission: Stage.GenericConfig.WIDGET_PERMISSION('sitesMap'),
+    categories: [Stage.GenericConfig.CATEGORY.SPIRE],
+    supportedEditions: [Stage.Common.Consts.licenseEdition.premium, Stage.Common.Consts.licenseEdition.spire],
+    initialConfiguration: [
+        Stage.GenericConfig.POLLING_TIME_CONFIG(10),
+        {
+            id: 'showAllLabels',
+            name: 'Show all the site labels',
+            default: false,
+            type: Stage.Basic.GenericField.BOOLEAN_TYPE
+        }
+    ],
+
+    fetchParams(_widget, toolbox) {
+        return {
+            blueprint_id: toolbox.getContext().getValue('blueprintId'),
+            id: toolbox.getContext().getValue('deploymentId')
+        };
+    },
+
+    fetchData(_widget, toolbox, params) {
+        const { DeploymentActions, SummaryActions } = Stage.Common;
+        const sitesWithNamesAndLocations: Promise<Stage.Types.PaginatedResponse<
+            Stage.Common.Map.SiteWithPosition
+        >> = new DeploymentActions(toolbox).doGetSitesNamesAndLocations();
+
+        const sitesSummary: Promise<Stage.Types.PaginatedResponse<SiteSummary>> = new SummaryActions(
+            toolbox
+        ).doGetDeployments('site_name', {
+            _include: 'id,site_name,deployment_status',
+            _sub_field: 'deployment_status',
+            ...params
+        });
+
+        return Promise.all([sitesWithNamesAndLocations, sitesSummary]).then(([sites, summary]) => {
+            const statusesSummaryLookupMap: Record<string, DeploymentStatusesSummary> = {};
+            const sitesData: SitesData = {};
+
+            const deploymentsWithSitesSummary = summary.items.filter(item => item.site_name);
+            deploymentsWithSitesSummary.forEach(siteSummary => {
+                statusesSummaryLookupMap[siteSummary.site_name] = getDeploymentStatusesSummary(siteSummary);
+            });
+
+            const sitesWithLocation = sites.items.filter(site => isFinite(site.latitude) && isFinite(site.longitude));
+            sitesWithLocation.forEach(site => {
+                sitesData[site.name] = {
+                    ...site,
+                    statusesSummary: statusesSummaryLookupMap[site.name] || emptyDeploymentStatusesSummary
+                };
+            });
+
+            return { sitesData, sitesAreDefined: sites.items.length > 0 };
+        });
+    },
+
+    render(widget, data, _error, toolbox) {
+        const { Loading } = Stage.Basic;
+
+        if (_.isEmpty(data)) {
+            return <Loading />;
+        }
+
+        const formattedData = data || { sitesData: {}, sitesAreDefined: false };
```

Comment:
I assume this default value for `formattedData` is there to guard against the empty state of `data`. If that's the case, you can use 

```suggestion
        if (Stage.Utils.isEmptyWidgetData(data)) {
            return <Loading />;
        }

        const formattedData = data;
```

but I am not certain if there are other cases when the default value is necessary :slightly_smiling_face:

Reply:
That was the problem (guard against the empty state of `data`). Good to know there's a function to cover that 🙂 Used it.